### ISSUE TITLE
AttributeError when query passed is a list and not a dict

### ISSUE THREAD

**Issue by Amangiri99** [strawberry-graphql/strawberry#3398]
While doing a testing my graphql endpoints, I came across an issue that if the query sent in the request is a list an attributeError is raised.
On further investigation I found out when parsing the request data to [GrahpqlRequestData](https://github.com/strawberry-graphql/strawberry/blob/main/strawberry/http/sync_base_view.py#L159)  we are accessing the data assuming it's a dict, rather we should validate it first if data is a valid dictionary

## System Information

 - Operating system: MacOS
 - Strawberry version (if applicable): 0.219.2

**Comment by patrick91** [comment]
Hi @Amangiri99 can you add a reproduction? 😊 thanks!

**Comment by Amangiri99** [comment]
Hit any of the graphql endpoints, by sending a query. The query shouldn't be a dict rather a list.
Example: --data “[{\”query\":\"{connectionId(id:\\\"0e0fac6d-2d7c-4de2-bdcd-586c48ef8c2f\\\"){getDataSource{DataSource}}}\”}]” \

Also I got this error while making request using curl command.

**Comment by patrick91** [comment]
@Amangiri99 how come you're sending a list? usually that's done when you want to run multiple operations (like batching)

**Comment by patrick91** [comment]
@Amangiri99 yeah, but you could send a list using things like https://www.apollographql.com/docs/react/api/link/apollo-link-batch-http/

I don't think we should worry about lists with single items for now (batching will help with that), but maybe we can improve the error?

**Comment by Amangiri99** [comment]
@patrick91  Right, we should improve the error. We should send a generic error if a list is received instead of object.

**Comment by patrick91** [comment]
Fixed here https://github.com/strawberry-graphql/strawberry/pull/3755
